### PR TITLE
remove unnecessary imports so WebGPU can be thinner

### LINKED ISSUE #15580: Support for WebGPUEngine
`WebGPUEngine` can't really be used until `initAsync` is complete. If something outside of the `Viewer` owns the engine instance, they are responsible for calling `initAsync`, but the `Viewer` still needs to await completion of `initAsync`. There doesn't seem to be a way to do this currently.

### PR DESCRIPTION
**Pull request by deltakosh**
Fixes #15580

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by deltakosh** [comment]
@sebavan  / @Popov72 / @RaananW : I removed the hard coded imports to Engine extensions and moved then back to the engine class. That way WebGPUEngine can have its own version and not the engine version + webgpu version.

@RaananW / @sebavan : I want to make sure this is not a problem for ThinEngine

**Comment by deltakosh** [comment]
Because I'm a GIT illiterate I also included the content from https://github.com/BabylonJS/Babylon.js/pull/15572

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15579/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15579/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15579/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15579/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15579/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15579/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15579/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15579/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15579/merge/testResults/webgl2playwright/index.html

File: packages/tools/viewer-alpha/src/viewerFactory.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // eslint-disable-next-line import/no-internal-modules
-import { RGBDTextureTools, type AbstractEngine, type AbstractEngineOptions, type EngineOptions, type WebGPUEngineOptions } from "core/index";
+import { type AbstractEngine, type AbstractEngineOptions, type EngineOptions, type WebGPUEngineOptions } from "core/index";
```
Comment (ryantrem) [inline packages/tools/viewer-alpha/src/viewerFactory.ts:2]: I think for the build this needs to be:
```ts
import type { AbstractEngine, AbstractEngineOptions, EngineOptions, WebGPUEngineOptions } from "core/index";
```
I think that was my mistake.

**Comment by RaananW** [comment]
Working on resolving the es6 build issue (filesize and chunks)

**Review by RaananW — APPROVED** [review]
I don't see it influencing thinengine, the only thing you added to abstractengine already exists there

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15579/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Pull request merged**